packages/gestalt-datepicker/src/DateRange/InternalDatePicker.js

```diff
@@ -0,0 +1,112 @@
+// @flow strict-local
+import {
+  type AbstractComponent,
+  type Element,
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import ReactDatePicker, { registerLocale } from 'react-datepicker';
+import { Icon } from 'gestalt';
+import styles from '../DatePicker.css';
+import { type Props } from '../DatePicker.js';
+
+type ModifiedProps = {|
+  ...Props,
+  onChange: ({| startDate: Date, endDate: Date |}) => void,
+|};
+
+const InternalDatePickerWithForwardRef: AbstractComponent<ModifiedProps, HTMLInputElement> =
+  forwardRef<ModifiedProps, HTMLInputElement>(function InternalDatePicker(
+    {
+      excludeDates,
+      id,
+      includeDates,
+      localeData,
+      maxDate,
+      minDate,
+      onChange,
+      rangeEndDate,
+      rangeStartDate,
+    }: ModifiedProps,
+    ref,
+  ): Element<'div'> {
+    const innerInputRef = useRef<null | HTMLInputElement>(null);
+    useImperativeHandle(ref, () => innerInputRef.current);
+
+    // We keep month in state to trigger a re-render when month changes since height will vary by where days fall
+    // in the month and we need to keep the popover pointed at the input correctly
+    const [, setMonth] = useState<?number>();
+    const [format, setFormat] = useState<?string>();
+    const [updatedLocale, setUpdatedLocale] = useState<?string>();
+    const [initRangeHighlight, setInitRangeHighlight] = useState<?Date>();
+
+    useEffect(() => {
+      setInitRangeHighlight(rangeStartDate || rangeEndDate);
+    }, [rangeStartDate, rangeEndDate]);
+
+    useEffect(() => {
+      if (localeData && localeData.code) {
+        registerLocale(localeData.code, localeData);
+        setUpdatedLocale(localeData.code);
+        setFormat(
+          localeData?.formatLong
+            ?.date({ width: 'short' })
+            .replace(/(y{1,4})/, 'yyyy')
+            .replace(/(d{1,2})/, 'dd')
+            .replace(/(M{1,2})/, 'MM') ?? 'MM/dd/yyyy',
+        );
+      }
+    }, [localeData]);
+
+    return (
+      <div className="_gestalt">
```

Comment:
Can we give this a min/max-height? It's kind of jarring when the height changes when the calendar is moved from left to right. 

I think it should always be 6 rows to be safe:

<img width="605" alt="image" src="https://github.com/pinterest/gestalt/assets/5509813/4db8b83c-e58f-4f5c-9332-715fa910dcc1">

Reply:
Discussed already with designers. Decided to leave as is.